### ISSUE TITLE
Implementing a linear regression model to predict the prices of houses based on their square footage and the number of bedrooms and bathrooms.

### ISSUE THREAD

**Issue by aryan-18-glitch** [aryan-18-glitch/PRODIGY_ML_01#1]
🏡 House Price Prediction using Linear Regression
This project demonstrates how to build a simple yet effective linear regression model to predict house prices based on key features such as square footage, number of bedrooms, and number of bathrooms. It's an ideal starting point for beginners looking to understand the basics of machine learning and data-driven prediction in real estate.

📌 Features
Predicts house prices using linear regression

Uses key property features: square footage, bedrooms, bathrooms

Clean and minimal code using Python, pandas, and scikit-learn

Includes data preprocessing, model training, evaluation, and prediction

📊 Technologies Used
Python

Pandas for data handling

Scikit-learn for model building and evaluation

Matplotlib/Seaborn for optional data visualization

🚀 Use Cases
Educational: Learn how regression works with real-world-style data

Prototyping: Build a basic house price estimation tool

Extendable: Add more features like location, age, or amenities for improved accuracy

**Comment by aryan-18-glitch** [comment]
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

# Load the training data
train_data = pd.read_csv("Data/house-prices-advanced-regression-techniques/train.csv")
print(train_data.head())

# Define features and target variable
features = ['GrLivArea', 'BedroomAbvGr', 'FullBath']
target = 'SalePrice'

# Check for missing values in the features
if train_data[features].isnull().any().any():
    print("Warning: There are missing values in the features.")
    # You may want to handle missing values here

# Prepare the feature matrix (X) and target vector (y)
x = train_data[features]
y = train_data[target]

# Split the data into training and testing sets
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42)

# Create and train the linear regression model
model = LinearRegression()
model.fit(x_train, y_train)

# Make predictions on the test set
y_pred = model.predict(x_test)

# Calculate evaluation metrics
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

# Print evaluation metrics
print(f'Mean Squared Error: {mse}')
print(f'R-squared: {r2}')

# Predict the price for a new house
new_data = pd.DataFrame([[2000, 3, 2]], columns=['GrLivArea', 'BedroomAbvGr', 'FullBath'])
predicted_price = model.predict(new_data)
print(f'Predicted price for the new house: ${predicted_price[0]:,.2f}')
